torch/distributed/checkpoint/state_dict.py

```diff
@@ -683,11 +693,33 @@ def _load_optim_state_dict(
                 optim_state_dict = FSDP.optim_state_dict_to_load(
                     model, optim, optim_state_dict
                 )
+        elif info.broadcast_from_rank0:
+            info.full_state_dict = False
+            local_state_dict = _get_optim_state_dict(model, (optim,), info)
```

Comment:
are we using `_get_optim_state_dict` instead of `torch.optim.Optimizer.state_dict` since we want to align FQNs between `local_state_dict` and `optim_state_dict` ? `torch.optim.Optimizer.state_dict` only give us ID keys

Reply:
yes, it is easier to proceed with keys.